### PR TITLE
287: Convert MoveProductFromShoppingCartToWishlistTest to MFTF

### PR DESCRIPTION
**Pull request by MilaLesechko**
**Test Case**
[Magento\Wishlist\Test\TestCase\MoveProductFromShoppingCartToWishlistTest](https://github.com/magento/magento-functional-tests-migration/tree/2.3-develop/dev/tests/functional/tests/app/Magento/Wishlist/Test/TestCase/MoveProductFromShoppingCartToWishlistTest.php)

**Covered variations:**
MoveProductFromShoppingCartToWishlistTestVariation2
MoveProductFromShoppingCartToWishlistTestVariation4
MoveProductFromShoppingCartToWishlistTestVariation5
MoveProductFromShoppingCartToWishlistTestVariation6

### Fixed Issues (if relevant)

1. magento/magento-functional-tests-migration#287:  Convert MoveProductFromShoppingCartToWishlistTest to MFTF

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontMoveVirtualProductFromShoppingCartToWishlistTest.xml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            <actionGroup ref="logout" stepKey="logout"/>
+        </after>
+
+        <!-- 1. Login as a customer -->
+        <actionGroup ref="LoginToStorefrontActionGroup" stepKey="loginToStorefrontAccount">
+            <argument name="Customer" value="$$createCustomer$$"/>
+        </actionGroup>
+
+        <!-- Open Virtual Product page -->
+        <amOnPage url="{{StorefrontProductPage.url($$createProduct.custom_attributes[url_key]$$)}}" stepKey="OpenStoreFrontProductPage"/>
```
Comment (lbajsarowicz) [inline app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontMoveVirtualProductFromShoppingCartToWishlistTest.xml]: Does necessarily to be PascalCase, use snailCase instead :-)

File: dev/tests/functional/tests/app/Magento/Wishlist/Test/TestCase/MoveProductFromShoppingCartToWishlistTest.xml [hunk 2]
```diff
@@ -29,21 +30,23 @@
             <constraint name="Magento\Wishlist\Test\Constraint\AssertProductDetailsInWishlist" />
         </variation>
         <variation name="MoveProductFromShoppingCartToWishlistTestVariation4">
-            <data name="tag" xsi:type="string">test_type:extended_acceptance_test</data>
+            <data name="tag" xsi:type="string">test_type:extended_acceptance_test, mftf_migrated:yes</data>
```
Comment (lbajsarowicz) [inline dev/tests/functional/tests/app/Magento/Wishlist/Test/TestCase/MoveProductFromShoppingCartToWishlistTest.xml]: I don't think we need space there.

File: app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontMoveDynamicBundleProductFromShoppingCartToWishlistTest.xml [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        <!-- Select Mini Cart and select 'View And Edit Cart' -->
+        <actionGroup ref="clickViewAndEditCartFromMiniCart" stepKey="selectViewAndEditCart"/>
+
+        <!-- Assert move product to wishlist success message -->
+        <click selector="{{CheckoutCartProductSection.moveToWishlistByProductName($$createBundleProduct.name$$)}}" stepKey="moveToWishlist"/>
+        <waitForPageLoad stepKey="waitForPageLoad"/>
+        <see userInput="$$createBundleProduct.name$$ has been moved to your wish list." selector="{{CheckoutCartMessageSection.successMessage}}" stepKey="assertMoveProductToWishlistSuccessMessage"/>
+
+        <!-- Assert product is present in wishlist -->
+        <amOnPage url="{{StorefrontCustomerWishlistPage.url}}" stepKey="goToWishlistPage"/>
```
Comment (soumyau) [inline app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontMoveDynamicBundleProductFromShoppingCartToWishlistTest.xml]: steps 95-98 can be moved to an action group called AssertProductIsPresentInWishList, I see this being a re-usable action group across tests.

File: app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontMoveDynamicBundleProductFromShoppingCartToWishlistTest.xml [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        <waitForPageLoad stepKey="waitForWishlistPage"/>
+        <waitForElement selector="{{StorefrontCustomerWishlistProductSection.ProductTitleByName($$createBundleProduct.name$$)}}" time="30" stepKey="assertWishlistProductName"/>
+        <see userInput="$100.00" selector="{{StorefrontCustomerWishlistProductSection.ProductPriceByName($$createBundleProduct.name$$)}}" stepKey="AssertWishlistProductPrice"/>
+
+        <!-- Assert product details in Wishlist -->
+        <moveMouseOver selector="{{StorefrontCustomerWishlistProductSection.ProductInfoByName($$createBundleProduct.name$$)}}" stepKey="wishlistMoveMouseOverProduct"/>
```
Comment (soumyau) [inline app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontMoveDynamicBundleProductFromShoppingCartToWishlistTest.xml]: steps 101-106 can be moved to an action group AssertProductDetailsInWishlist. The action group can be referenced in all tests where these steps are called in this PR.

File: app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontMoveFixedBundleProductFromShoppingCartToWishlistTest.xml [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+        <see userInput="$$createBundleProduct.name$$ has been moved to your wish list." selector="{{CheckoutCartMessageSection.successMessage}}" stepKey="assertMoveProductToWishlistSuccessMessage"/>
+
+        <!-- Assert product is present in wishlist -->
+        <amOnPage url="{{StorefrontCustomerWishlistPage.url}}" stepKey="goToWishlistPage"/>
+        <waitForPageLoad stepKey="waitForWishlistPage"/>
+        <waitForElement selector="{{StorefrontCustomerWishlistProductSection.ProductTitleByName($$createBundleProduct.name$$)}}" time="30" stepKey="assertWishlistProductName"/>
+        <see userInput="$101.23" selector="{{StorefrontCustomerWishlistProductSection.ProductPriceByName($$createBundleProduct.name$$)}}" stepKey="AssertWishlistProductPrice"/>
+
+        <!-- Assert product details in Wishlist -->
+        <moveMouseOver selector="{{StorefrontCustomerWishlistProductSection.ProductInfoByName($$createBundleProduct.name$$)}}" stepKey="wishlistMoveMouseOverProduct"/>
```
Comment (soumyau) [inline app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontMoveFixedBundleProductFromShoppingCartToWishlistTest.xml]: AssertProductDetailsInWishlist created (as per previous comment) can be called here.

File: app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontMoveFixedBundleProductFromShoppingCartToWishlistTest.xml [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+        <!-- Select Mini Cart and select 'View And Edit Cart' -->
+        <actionGroup ref="clickViewAndEditCartFromMiniCart" stepKey="selectViewAndEditCart"/>
+
+        <!-- Assert move product to wishlist success message -->
+        <click selector="{{CheckoutCartProductSection.moveToWishlistByProductName($$createBundleProduct.name$$)}}" stepKey="moveToWishlist"/>
+        <waitForPageLoad stepKey="waitForPageLoad"/>
+        <see userInput="$$createBundleProduct.name$$ has been moved to your wish list." selector="{{CheckoutCartMessageSection.successMessage}}" stepKey="assertMoveProductToWishlistSuccessMessage"/>
+
+        <!-- Assert product is present in wishlist -->
+        <amOnPage url="{{StorefrontCustomerWishlistPage.url}}" stepKey="goToWishlistPage"/>
```
Comment (soumyau) [inline app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontMoveFixedBundleProductFromShoppingCartToWishlistTest.xml]: AssertProductIsPresentInWishList created (as per previous comment) can be called here.

File: app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontMoveConfigurableProductFromShoppingCartToWishlistTest.xml [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+        <!-- Add product to the cart and Assert add product to cart success message-->
+        <actionGroup ref="addToCartFromStorefrontProductPage" stepKey="addToCartVirtualProductFromStorefrontProductPage">
+            <argument name="productName" value="$$createConfigProduct.name$$"/>
+        </actionGroup>
+
+        <!-- Select Mini Cart and select 'View And Edit Cart' -->
+        <actionGroup ref="clickViewAndEditCartFromMiniCart" stepKey="selectViewAndEditCart"/>
+
+        <!-- Assert move product to wishlist success message -->
+        <click selector="{{CheckoutCartProductSection.moveToWishlistByProductName($$createConfigProduct.name$$)}}" stepKey="moveToWishlist"/>
```
Comment (soumyau) [inline app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontMoveConfigurableProductFromShoppingCartToWishlistTest.xml]: steps 144-146 can be moved to an action group called AssertMoveProductToWishListSuccessMessage, I see this being a re-usable action group across tests.

File: app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontMoveConfigurableProductFromShoppingCartToWishlistTest.xml [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+        <see userInput="$$createConfigProduct.name$$ has been moved to your wish list." selector="{{CheckoutCartMessageSection.successMessage}}" stepKey="assertMoveProductToWishlistSuccessMessage"/>
+
+        <!-- Assert product is present in wishlist -->
+        <amOnPage url="{{StorefrontCustomerWishlistPage.url}}" stepKey="goToWishlistPage"/>
+        <waitForPageLoad stepKey="waitForWishlistPage"/>
+        <waitForElement selector="{{StorefrontCustomerWishlistProductSection.ProductTitleByName($$createConfigProduct.name$$)}}" time="30" stepKey="assertWishlistProductName"/>
+        <see userInput="$20.00" selector="{{StorefrontCustomerWishlistProductSection.ProductPriceByName($$createConfigProduct.name$$)}}" stepKey="AssertWishlistProductPrice"/>
+
+        <!-- Assert product details in Wishlist -->
+        <moveMouseOver selector="{{StorefrontCustomerWishlistProductSection.ProductInfoByName($$createConfigProduct.name$$)}}" stepKey="wishlistMoveMouseOverProduct"/>
```
Comment (soumyau) [inline app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontMoveConfigurableProductFromShoppingCartToWishlistTest.xml]: steps 155-160 can be moved to an action group AssertProductDetailsInWishlist. The action group can be referenced in all tests where these steps are called in this PR.

File: app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontMoveConfigurableProductFromShoppingCartToWishlistTest.xml [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+
+        <!-- Select Mini Cart and select 'View And Edit Cart' -->
+        <actionGroup ref="clickViewAndEditCartFromMiniCart" stepKey="selectViewAndEditCart"/>
+
+        <!-- Assert move product to wishlist success message -->
+        <click selector="{{CheckoutCartProductSection.moveToWishlistByProductName($$createConfigProduct.name$$)}}" stepKey="moveToWishlist"/>
+        <waitForPageLoad stepKey="waitForPageLoad"/>
+        <see userInput="$$createConfigProduct.name$$ has been moved to your wish list." selector="{{CheckoutCartMessageSection.successMessage}}" stepKey="assertMoveProductToWishlistSuccessMessage"/>
+
+        <!-- Assert product is present in wishlist -->
```
Comment (soumyau) [inline app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontMoveConfigurableProductFromShoppingCartToWishlistTest.xml]: steps 149-152 can be moved to an action group called AssertProductIsPresentInWishList, I see this being a re-usable action group across tests.

File: app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontMoveDynamicBundleProductFromShoppingCartToWishlistTest.xml [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+        <!-- Add product to the cart and Assert add product to cart success message-->
+        <actionGroup ref="addToCartFromStorefrontProductPage" stepKey="addToCartVirtualProductFromStorefrontProductPage">
+            <argument name="productName" value="$$createBundleProduct.name$$"/>
+        </actionGroup>
+
+        <!-- Select Mini Cart and select 'View And Edit Cart' -->
+        <actionGroup ref="clickViewAndEditCartFromMiniCart" stepKey="selectViewAndEditCart"/>
+
+        <!-- Assert move product to wishlist success message -->
+        <click selector="{{CheckoutCartProductSection.moveToWishlistByProductName($$createBundleProduct.name$$)}}" stepKey="moveToWishlist"/>
```
Comment (soumyau) [inline app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontMoveDynamicBundleProductFromShoppingCartToWishlistTest.xml]: AssertMoveProductToWishList created (as per previous comment) can be called here.

File: app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontMoveFixedBundleProductFromShoppingCartToWishlistTest.xml [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+
+        <!-- Add product to the cart and Assert add product to cart success message-->
+        <actionGroup ref="addToCartFromStorefrontProductPage" stepKey="addToCartVirtualProductFromStorefrontProductPage">
+            <argument name="productName" value="$$createBundleProduct.name$$"/>
+        </actionGroup>
+
+        <!-- Select Mini Cart and select 'View And Edit Cart' -->
+        <actionGroup ref="clickViewAndEditCartFromMiniCart" stepKey="selectViewAndEditCart"/>
+
+        <!-- Assert move product to wishlist success message -->
```
Comment (soumyau) [inline app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontMoveFixedBundleProductFromShoppingCartToWishlistTest.xml]: AssertMoveProductToWishListSuccessMessage created (as per previous comment) can be called here.

File: app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontMoveVirtualProductFromShoppingCartToWishlistTest.xml [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+
+        <!-- Add Virtual product to the cart and Assert add product to cart success message-->
+        <actionGroup ref="addToCartFromStorefrontProductPage" stepKey="addToCartVirtualProductFromStorefrontProductPage">
+            <argument name="productName" value="$$createProduct.name$$"/>
+        </actionGroup>
+
+        <!-- Select Mini Cart and select 'View And Edit Cart' -->
+        <actionGroup ref="clickViewAndEditCartFromMiniCart" stepKey="selectViewAndEditCart"/>
+
+        <!-- Assert move product to wishlist success message -->
```
Comment (soumyau) [inline app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontMoveVirtualProductFromShoppingCartToWishlistTest.xml]: same as previous comments.

File: app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontMoveVirtualProductFromShoppingCartToWishlistTest.xml [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+        <click selector="{{CheckoutCartProductSection.moveToWishlistByProductName($$createProduct.name$$)}}" stepKey="moveToWishlist"/>
+        <waitForPageLoad stepKey="waitForPageLoad"/>
+        <see userInput="$$createProduct.name$$ has been moved to your wish list." selector="{{CheckoutCartMessageSection.successMessage}}" stepKey="assertMoveProductToWishlistSuccessMessage"/>
+        
+        <!-- Assert product is present in wishlist -->
+        <amOnPage url="{{StorefrontCustomerWishlistPage.url}}" stepKey="goToWishlistPage"/>
```
Comment (soumyau) [inline app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontMoveVirtualProductFromShoppingCartToWishlistTest.xml]: same as previous comments.

File: app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontMoveDynamicBundleProductFromShoppingCartToWishlistTest.xml (hunk 3 above)
Comment (tomreece) [reply app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontMoveDynamicBundleProductFromShoppingCartToWishlistTest.xml]: I fixed this in our repo.

File: app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontMoveDynamicBundleProductFromShoppingCartToWishlistTest.xml (hunk 4 above)
Comment (tomreece) [reply app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontMoveDynamicBundleProductFromShoppingCartToWishlistTest.xml]: I fixed this in our repo.

File: app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontMoveFixedBundleProductFromShoppingCartToWishlistTest.xml (hunk 5 above)
Comment (tomreece) [reply app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontMoveFixedBundleProductFromShoppingCartToWishlistTest.xml]: I fixed this in our repo.

File: app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontMoveFixedBundleProductFromShoppingCartToWishlistTest.xml (hunk 6 above)
Comment (tomreece) [reply app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontMoveFixedBundleProductFromShoppingCartToWishlistTest.xml]: I fixed this in our repo.

File: app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontMoveConfigurableProductFromShoppingCartToWishlistTest.xml (hunk 7 above)
Comment (tomreece) [reply app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontMoveConfigurableProductFromShoppingCartToWishlistTest.xml]: I fixed this in our repo.

**Comment by tomreece** [comment]
This PR is in the queue to be delivered to mainline.

PLEASE DO NOT MAKE ANY ADDITIONAL CHANGES

**Pull request merged**